src/rdb.c

```diff
@@ -1270,22 +1271,17 @@ int rdbSaveRio(rio *rdb, int *error, int rdbflags, rdbSaveInfo *rsi) {
                 aofReadDiffFromParent();
             }
 
-            /* Update COW info every 1 second (approximately).
+            /* Update child info every 1 second (approximately).
              * in order to avoid calling mstime() on each iteration, we will
              * check the diff every 1024 keys */
-            if ((key_count & 1023) == 0) {
-                key_count = 0;
+            if ((key_count++ % 1024) == 0) {
```

Comment:
sorry for a long comment about nonsense, i guess i take this (divisions) personally.

there was indeed no need to reset `key_count` even in the old code.
i.e. if the code would have done `> 1023` it would have needed to reset the counter, but using either `&1023` or `%1024` would be ok even if we don't reset it.

however, i'd rather revert this to use `&` since divisions gives me the shivers, in this case the compiler will convert it to an `&1023`, but i'm afraid someone will some day change `1024` with another number and then we'll have division.

i know it's a micro optimization that we'll never be able to measure, but personally i think doing `%1000` instead of `&1023` is a bad practice.

Reply:
Thanks for your explanation, Done.